docs/GraFx-Developers/connectors/media-connector/add-variable-settings-to-your-connector/index.md

```diff
@@ -0,0 +1,231 @@
+6. Click "Select image" to observe how the number of displayed images changes based on your setting.
+
+## Adding a Rectangle/Square Image Setting
+
+
+The picsum.photos API supports both outputting rectangular and square images. We can enhance our connector by adding a setting that allows Designes to choose between these two image crops. This setting will affect how images are displayed in both the image selector and the document frame.
+
+### Step 1: Update `getConfigurationOptions`
+
+```typescript
+  getConfigurationOptions(): Connector.ConnectorConfigValue[] | null {
+    return [
+      {
+        name: "limit",
+        displayName: "Number (1 to 100) of Images to Display",
+        type: "text"
+      },
+      {
+        name: "wide",
+        displayName: "Display images as rectangluar instead of square",
+        type: "boolean"
+      }
+    ];
+  }
+  ```
+
+This addition introduces a new boolean option named "wide" that users can toggle to switch between rectangular and square image displays.
+
+### Step 2: Update the `download` Method
+
+Next, we'll update our `download` method to incorporate this new setting:
+
+```typescript
+  async download(
+    id: string,
+    previewType: Media.DownloadType,
+    intent: Media.DownloadIntent,
+    context: Connector.Dictionary
+  ): Promise<Connector.ArrayBufferPointer> {
+
+    // Check to see if we are a thumbnail in the UI or being used in another situation.
+    switch (previewType) {
+      case "thumbnail": {
+        const picture = await this.runtime.fetch(`https://picsum.photos/id/${id}/${(context.wide) ? "400/" : ""}200`, { method: "GET" });
+        return picture.arrayBuffer;
+      }
+      default: {
+        const picture = await this.runtime.fetch(`https://picsum.photos/id/${id}/${(context.wide) ? "2000/" : ""}1000`, { method: "GET" });
+        return picture.arrayBuffer;
+      }
+    }
+  }
+```
+
+Key changes in this method:
+1. We now use the `context.wide` setting to determine the image dimensions.
+2. For thumbnails, we use 400x200 for wide (rectangular) images and 200x200 for square images.
+3. For full-size images, we use 2000x1000 for wide images and 1000x1000 for square images.
+
+### Step 3: Publish and Test
+
+The "wide" setting controls how the image is loaded into the frame. To best observe this effect:
+
+1. Publish your updated connector using the method described in the previously.
+2. In your document, add or select an existing image variable using this Connector.
+3. Set up an image frame with the mode set to "Fit" and linked to the variable.
+4. Locate the new "Display images as rectangular instead of square" setting in the connector options.
+5. Toggle this setting and observe how it affects the images in both the selector and the document frame.
+
+This demonstrates that our connector is not just affecting the image selection process, but also how the image is displayed within the document itself.
+
+### Key Accomplishments
```

Comment:
This should all be in the add-environment-options doc?

Also, the next steps link at the bottom of this page goes to building connector with auth (not there yet, already know that), it should have a link to the environment options doc instead as next step?

Reply:
This section needs to be moved to environment options